pkg/clusteragent/admission/mutate/config_test.go

```diff
@@ -116,26 +116,32 @@ func TestInjectHostIP(t *testing.T) {
 	pod = withLabels(pod, map[string]string{"admission.datadoghq.com/enabled": "true"})
 	err := injectConfig(pod, "", nil)
 	assert.Nil(t, err)
-	assert.Equal(t, pod.Spec.Containers[0].Env[0].Name, "DD_AGENT_HOST")
-	assert.Equal(t, pod.Spec.Containers[0].Env[0].ValueFrom.FieldRef.FieldPath, "status.hostIP")
+	envs := pod.Spec.Containers[0].Env
+	lastEnv := envs[len(envs)-1]
```

Comment:
Something to consider:  this test is strongly coupled to the order of injected env vars.  Perhaps it would be cleaner to lookup the element from `Env` which matches the expected `Name`, and assert on its value.  This would be (in my opinion) an incremental improvement in the stability of these tests.

Reply:
Agreed, feel free to do so :)

Reply:
Done.  Although now I'm left wondering whether we actually *do* care about order now, since our env vars are supposed to precede the other ones.